File: app/models/dummy/OrderModule.java
package models.dummy;

import models.*;

import java.util.*;

/**
 * Created by Beemen on 04/08/2015.
 */
public class OrderModule implements IOrderModule {

    private final HashMap<IEndUser,List<IOrder>>_Orders = new HashMap<>();

    @Override
    public StandardReturn SubmitOrder(IOrder order, IEndUser user) {
        if(order.Archives().size() > 0){
            order.IssueDate(new Date());

            _Orders.putIfAbsent(user, new ArrayList<>());
            _Orders.get(user).add(order);

            return StandardReturn.OK();
        }
        else {
            return StandardReturn.InvalidInput("");
        }
    }

    @Override
    public List<IOrder> GetUserOrders(IEndUser user) {
        return _Orders.getOrDefault(user, new ArrayList<>());
    }
}

File: Models/dummy/OrderModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using end_user_gui.Modules;

namespace end_user_gui.Models.dummy
{
    public class OrderModule : IOrderModule
    {
        private readonly Dictionary<EndUser, List<IOrder>> _Orders = new Dictionary<EndUser, List<IOrder>>();

        public StandardReturn SubmitOrder(IOrder order, EndUser user)
        {
            if (order.Archives.Count > 0)
            {
                order.IssueDate = DateTime.Now;

                if (!_Orders.ContainsKey(user))
                    _Orders[user] = new List<IOrder>();

                _Orders[user].Add(order);

                return StandardReturn.OK();
            }
            else
            {
                return StandardReturn.InvalidInput("");
            }
        }

        public List<IOrder> GetUserOrders(EndUser user)
        {
            return _Orders.ContainsKey(user) ? _Orders[user] : new List<IOrder>();
        }
    }
}